core/app/models/spree/promotion/rules/user.rb

```diff
@@ -6,7 +6,7 @@ class User < PromotionRule
 
         has_many :promotion_rule_users, class_name: 'Spree::PromotionRuleUser',
                                         foreign_key: :promotion_rule_id
-        has_many :users, through: :promotion_rule_users, class_name: ::Spree.user_class.to_s
+        has_many :users, through: :promotion_rule_users, class_name: "::#{Spree.user_class.to_s}"
```

Comment:
Hound, I love you. The redundant use of `to_s` reduces the failure rate by a lot here. Not that I understand why that's the case, but it seems to be.